domain/application/service/unit.go

```diff
@@ -341,15 +335,49 @@ func (s *Service) GetUnitDisplayStatus(ctx context.Context, unitName coreunit.Na
 	if err != nil {
 		return nil, errors.Trace(err)
 	}
+
 	containerStatus, err := s.st.GetUnitCloudContainerStatus(ctx, unitUUID)
-	if errors.Is(err, applicationerrors.UnitStatusNotFound) {
-		return unitDisplayStatus(workloadStatus, nil)
-	} else if err != nil {
+	if err != nil && !errors.Is(err, applicationerrors.UnitStatusNotFound) {
 		return nil, errors.Trace(err)
 	}
 	return unitDisplayStatus(workloadStatus, containerStatus)
 }
 
+// GetUnitAndAgentDisplayStatus returns the unit and agent display status of the
+// specified unit. The display status a function of both the unit workload status
+// and the cloud container status. It returns an error satisfying
+// [applicationerrors.UnitNotFound] if the unit doesn't exist.
+func (s *Service) GetUnitAndAgentDisplayStatus(ctx context.Context, unitName coreunit.Name) (agent *corestatus.StatusInfo, workload *corestatus.StatusInfo, _ error) {
+	if err := unitName.Validate(); err != nil {
+		return nil, nil, errors.Trace(err)
+	}
+
+	// TODO (stickupkid) This should just be 1 or 2 calls to the state layer
+	// to get the agent and workload status.
+
+	unitUUID, err := s.st.GetUnitUUIDByName(ctx, unitName)
+	if err != nil {
+		return nil, nil, errors.Trace(err)
+	}
+
+	agentStatus, err := s.st.GetUnitAgentStatus(ctx, unitUUID)
+	if err != nil {
+		return nil, nil, errors.Trace(err)
+	}
+
+	workloadStatus, err := s.st.GetUnitWorkloadStatus(ctx, unitUUID)
+	if err != nil {
+		return nil, nil, errors.Trace(err)
+	}
+
+	containerStatus, err := s.st.GetUnitCloudContainerStatus(ctx, unitUUID)
+	if err != nil && !errors.Is(err, applicationerrors.UnitStatusNotFound) {
+		return nil, nil, errors.Trace(err)
+	}
+
+	return decodeUnitAgentWorkloadStatus(agentStatus, workloadStatus, containerStatus)
```

Comment:
This as well I think should be re-named to mention we're decoding into the display status